src/exception/Assert.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <String/BsString.h>
+
+#include <exception/AssertionException.hpp>
+#include <exception/Throw.hpp>
+#include <log/logging.hpp>
+
```

Comment:
Can you include some usage examples in a comment here? Also the info that assertions might be transformed into error-logs in some build configurations would fit in nicely here.

(Don't explicitly say which build configs are affected, otherwise it's just another place to fix if we ever change that)